python-package/src/promptguard/promptguard_service.py

```diff
@@ -103,37 +109,34 @@ def _send_request_to_promptguard_service(
         if the request was successful
     """
 
+    global _session
+    global _sessionLock
+
+    with _sessionLock:
+        if _session is None:
+            _session = requests.Session()
+            _session.mount("httpa://", HTTPAAdapter(_get_default_validators()))
+
     api_key = get_api_key()
     hostname, port = get_server_config()
 
-    ctx = AttestedTLSContext(_get_default_validators())
-    conn = AttestedHTTPSConnection(hostname, ctx, port)
+    response = _session.request(
```

Comment:
This code doesn't handle the case where connection was closed (maybe by remote). Possibly need something like https://github.com/singer-io/tap-skubana/pull/6/files or better.